### PR TITLE
[Chore] Add unit test to getSurveyById

### PR DESCRIPTION
**Pull request by sleepylee**
## What happened 👀
Just continue to add more unit test coverage: SurveyRepository > getSurveyById
 
## Insight 📝
- Write test to validate responses
- Checking for the extension/computed attribute like `hdCoverImage`...
 
## Proof Of Work 📹
Codecov ++

### CODE REVIEW

**Review by manh-t — APPROVED** [review]
I really like the tree map, lol

File: test/test_resources/single_survey.json [hunk 1]
```diff
@@ -0,0 +1,378 @@
+{
```
Comment (luongvo) [inline test/test_resources/single_survey.json:1]: thank you for preparing test json resouces for my IC 💁‍♂️

File: test/repositories/survey_repository_test.dart [hunk 2]
```diff
@@ -58,4 +59,48 @@ main() async {
...
+      when(mockGraphQLClient.query(any)).thenAnswer(
+        (_) async => QueryResult.optimistic(
+          data: singleSurveyData,
+        ),
+      );
+      final result = await testedSurveyRepository.getSurveyById("any");
+
+      expect(result.hdCoverImageUrl, result.coverImageUrl + "l");
+      expect(result.questions.first.hdCoverImageUrl,
+          result.questions.first.coverImageUrl + "l");
```
Comment (luongvo) [inline test/repositories/survey_repository_test.dart:93]: can we use `hdCoverImageUrl` instead? 🤓

**Review by luongvo — COMMENTED** [review]
rest lgtm

**Review by markgravity — APPROVED** [review]
LGTM

File: test/repositories/survey_repository_test.dart (hunk 2 above)
Comment (sleepylee) [reply test/repositories/survey_repository_test.dart:93]: hdCoverImageUrl is a computed attribute, so we are trying to test its logic against the original data 🤔

Comment (luongvo) [reply test/repositories/survey_repository_test.dart:93]: to test its logic, better to test it on survey_test class instead?

**Pull request merged**